drivers/stepper/step_dir/Kconfig

```diff
@@ -0,0 +1,32 @@
+# Copyright (c) 2024 Fabian Blatz <fabianblatz@gmail.com>
+# SPDX-License-Identifier: Apache-2.0
+
+config STEP_DIR_STEPPER
+	bool
+	help
+	  Enable library used for step direction stepper drivers.
+
+config STEP_DIR_STEPPER_GENERATE_ISR_SAFE_EVENTS
+	bool "Guarantee non ISR callbacks upon stepper events"
+	depends on STEP_DIR_STEPPER
+	help
+	  Enable the dispatch of stepper generated events via
+	  a message queue to guarantee that the event handler
+	  code is not run inside of an ISR. Can be disabled, but
+	  then registered stepper event callback must be ISR safe.
+
+config STEP_DIR_STEPPER_EVENT_QUEUE_LEN
+	int "Maximum number of pending stepper events"
+	default 4
+	depends on STEP_DIR_STEPPER_GENERATE_ISR_SAFE_EVENTS
+	help
+	  The maximum number of stepper events that can be pending before new events
+	  are dropped.
+
+config STEP_DIR_STEPPER_COUNTER_TIMING
```

Comment:
I think the idea was to go via device tree instead of Kconfig, in order to be able to select on per instance basis. https://github.com/zephyrproject-rtos/zephyr/pull/82843#issuecomment-2538045491

Reply:
Yeah I added the Kconfig to remove the source file from compilation if you are sure you only need the delayed work.